src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int factCapacity;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        factCapacity = INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
+        checkSize();
+        int hash = hash(key);
+        int bucketIndex = hash % factCapacity;
+
+        if (table[bucketIndex] == null) {
+            table[bucketIndex] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K, V> node = table[bucketIndex];
 
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+
+            while (node.next != null) {
+                if (node.hash == hash && Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                }
+                node = node.next;
+            }
+
+            if (node.hash == hash && Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+
+            node.next = new Node<>(hash, key, value, null);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int hash = hash(key);
+        int bucketIndex = hash % factCapacity;
+        Node<K, V> tmpNode = table[bucketIndex];
```

Comment:
Combine into one method
```suggestion
        int bucketIndex = getKeyIndex(key);
        Node<K, V> tmpNode = table[bucketIndex];
```